Google.Api.Generator/Generation/MetadataGenerator.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Google.Gapic.Metadata;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+
+namespace Google.Api.Generator.Generation
+{
+    internal static class MetadataGenerator
+    {
+        private const string GapicMetadataSchemaVersion = "1.0";
+        private const string GapicMetadataFileDescription = "This file maps proto services/RPCs to the corresponding library clients/methods";
+        private const string GapicMetadataLanguage = "csharp";
+
+        private const string TransportKeyGrpc = "grpc";
+
+        internal static string GenerateGapicMetadataJson(List<ServiceDetails> allServiceDetails)
+        {
+            
+            var gapicMetadata = new GapicMetadata
+            {
+                Schema = GapicMetadataSchemaVersion,
+                Comment = GapicMetadataFileDescription,
+                Language = GapicMetadataLanguage,
+                ProtoPackage = allServiceDetails.First().Package,
+                LibraryPackage = allServiceDetails.First().Namespace
+            };
+
+            foreach (var serviceDetails in allServiceDetails)
+            {
+                gapicMetadata.Services[serviceDetails.ServiceName] = ServiceForTransportMetadata(serviceDetails);
+            }
+
+            var gapicMetadataJson = JsonSerializer.Serialize(
+                gapicMetadata,
+                new JsonSerializerOptions
+                {
+                    WriteIndented = true,
+                    PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                    DictionaryKeyPolicy = JsonNamingPolicy.CamelCase
+                });
+
+            return gapicMetadataJson + "\n"; //trailing file newline to please github
+        }
+
+        private static GapicMetadata.Types.ServiceForTransport ServiceForTransportMetadata(ServiceDetails serviceDetails)

```

Comment:
nit: Add the `using GapicMetadata.Types` instead of fully qualifying the name here? And in the other method as well.

Reply:
In this case, `using static Google.Gapic.Metadata.GapicMetadata.Types;`
(`Types` is a class nested within `GapicMetadata`.)

Reply:
👍